pkg/ingester/ingester.go

```diff
@@ -963,7 +963,9 @@ func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*corte
 		// has sorted labels once hit the ingester).
 
 		// Look up a reference for this series.
-		ref, copiedLabels := app.GetRef(cortexpb.FromLabelAdaptersToLabels(ts.Labels))
+		tsLabels := cortexpb.FromLabelAdaptersToLabels(ts.Labels)
+		tsLabelsHash := tsLabels.Hash()
+		ref, copiedLabels := app.GetRef(tsLabels,tsLabelsHash)
```

Comment:
This seems not compilable.

Reply:
because this is related with vendor/prometheus . don't know how to match these two projects' codes. 
BTW, the PR of prometheus  is pendding.
https://github.com/prometheus/prometheus/pull/11485